code/lib/cli/src/add.ts

```diff
@@ -15,43 +10,23 @@ import { getStorybookVersion } from './utils';
 
 const logger = console;
 
-const LEGACY_CONFIGS = ['addons', 'config', 'presets'];
-
-const postinstallAddon = async (addonName: string, isOfficialAddon: boolean) => {
-  let skipMsg = null;
-  if (!isOfficialAddon) {
-    skipMsg = 'unofficial addon';
-  } else if (!fs.existsSync('.storybook')) {
-    skipMsg = 'no .storybook config';
-  } else {
-    skipMsg = 'no codmods found';
-    LEGACY_CONFIGS.forEach((config) => {
-      try {
-        const codemod = require.resolve(
-          // @ts-expect-error (it is broken)
-          `${getPackageName(addonName, isOfficialAddon)}/postinstall/${config}.js`
-        );
-        commandLog(`Running postinstall script for ${addonName}`)();
-        let configFile = path.join('.storybook', `${config}.ts`);
-        if (!fs.existsSync(configFile)) {
-          configFile = path.join('.storybook', `${config}.js`);
-          if (!fs.existsSync(configFile)) {
-            fs.writeFileSync(configFile, '', 'utf8');
-          }
-        }
-        spawnSync('npx', ['jscodeshift', '-t', codemod, configFile], {
-          stdio: 'inherit',
-          shell: true,
-        });
-        skipMsg = null;
-      } catch (err) {
-        // resolve failed, skip
-      }
-    });
-  }
+const postinstallAddon = async (addonName: string) => {
+  try {
+    const modulePath = require.resolve(`${addonName}/postinstall`, { paths: [process.cwd()] });
+    // eslint-disable-next-line import/no-dynamic-require, global-require
+    const postinstall = require(modulePath);
 
-  if (skipMsg) {
-    commandLog(`Skipping postinstall for ${addonName}, ${skipMsg}`)();
+    try {
+      logger.log(`Running postinstall script for ${addonName}`);
+      await postinstall();
+    } catch (e) {
+      logger.error(`Error running postinstall script for ${addonName}`);
+      logger.error(e);
+    }
+  } catch (e) {
+    // no postinstall script
+    logger.log(`No postinstall script for ${addonName}`);
```

Comment:
I don't think we need to log anything in this case. Will be pretty noisy since for now most addons won't have one.